### PR TITLE
Use upstream file packager instead of fork

### PR DESCRIPTION
**Pull request by dalcde**
This removes PYODIDE_PACKAGE_ABI and the corresponding ABI checks. We may introduce new ways to ensure packages are compiled for the right pyodide version in the future.

There is documentation yet to be updated. Addresses #990

### CODE REVIEW

**Review by rth — COMMENTED** [review]
Thanks @dalcde. One comment otherwise LGTM. 

> Addresses #990

You need to use key words "closes" or "fixes" for Github to auto-close the issue (full list in https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

File: Makefile.envs [hunk 1]
```diff
@@ -17,6 +17,3 @@ export PYTHONINCLUDE=$(PYODIDE_ROOT)/cpython/installs/python-$(PYVERSION)/includ
 
 # Use env variable if defined, otherwise fallback to './'
 export PYODIDE_BASE_URL?=./
-# This env variable is used to detect pyodide at build time,
-# do not rename it.
-export PYODIDE_PACKAGE_ABI=1
```
Comment (rth) [inline Makefile.envs:22]: (My comment didn't seem to be saved, so re-writing it).

The issue with removing this env variable, is that we have contributed fixes with it to upstream packages (at least scikit-image and scikit-learn) as a way to detect building it pyodide. Maybe we could keep it here for now, with a comment and open a separate issue on how to properly detect building in Pyodide (also taking into account that we might use cross-compilation in the future). Using `PYODIDE_BASE_URL` would be OK for this PR, but I'm not sure it's a good solution long term.

**Comment by dalcde** [comment]
>> Addresses #990
>
>You need to use key works "closes" or "fixes" for Github to auto-close the issue (full list in https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

I'm trying to avoid that so that I can repurpose #990 to tell us to
re-introduce versioning somehow

File: Makefile.envs (hunk 1 above)
Comment (dalcde) [reply Makefile.envs:22]: We could introduce a `PYODIDE` environment variable specifically for this purpose

File: Makefile.envs [hunk 2]
```diff
@@ -17,6 +17,9 @@ export PYTHONINCLUDE=$(PYODIDE_ROOT)/cpython/installs/python-$(PYVERSION)/includ
 
 # Use env variable if defined, otherwise fallback to './'
 export PYODIDE_BASE_URL?=./
-# This env variable is used to detect pyodide at build time,
-# do not rename it.
+
+# This environment variable is used for packages to detect if they are built
+# for pyodide during build time
+export PYODIDE=1
```
Comment (rth) [inline Makefile.envs:23]: OK, sounds good. Thanks!

**Pull request merged**